pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +71,109 @@ bc:
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.ErrorSeverity,
+						Key:      MissingInputImageStreamError,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+						Suggestion: osgraph.Suggestion(fmt.Sprintf("create the image stream %s with the required images", imageStreamNode.ResourceString())),
+					})
+
+					continue
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamTagNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					if _, ok := imageStream.Status.Tags[tagNode.ImageTag()]; !ok {
+						markers = append(markers, osgraph.Marker{
+							Node: bcNode,
+							RelatedNodes: []graph.Node{istNode,
+								imageStreamNode},
+
+							Severity: osgraph.WarningSeverity,
+							Key:      MissingImageStreamTagWarning,
+							Message: fmt.Sprintf("%s builds from %s, but the image stream tag does not exist.",
+								bcNode.(*buildgraph.BuildConfigNode).ResourceString(), istNode.(*imagegraph.ImageStreamTagNode).ResourceString()),
+							Suggestion: osgraph.Suggestion(fmt.Sprintf("examine analysis of build config outputs from this command and see if they build %s", istNode.(*imagegraph.ImageStreamTagNode).ResourceString())),
+						})
+					}
+					continue
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamImageNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					// cycle through the imageStream.Status.Tags.[]TagEvent.DockerImageReference and see if the image id is present
+					var found bool
+					var imageID string
+					for _, tagEventList := range imageStream.Status.Tags {
+						_, imageID, ok := imageapi.SplitImageStreamImage(tagNode.Name)
+						if ok {
+							for _, tagEvent := range tagEventList.Items {
+								if strings.Contains(tagEvent.DockerImageReference, imageID) {
+									found = true
+									break
+								}
+							}
+						}
+						if found {
+							break
+						}
+					}
+					if !found {
+						markers = append(markers, osgraph.Marker{
+							Node: bcNode,
+							RelatedNodes: []graph.Node{istNode,
+								imageStreamNode},
+
+							Severity: osgraph.WarningSeverity,
+							Key:      MissingImageStreamTagWarning,
+							Message: fmt.Sprintf("%s builds from %s, but the image stream image does not exist.",
+								bcNode.(*buildgraph.BuildConfigNode).ResourceString(), istNode.(*imagegraph.ImageStreamImageNode).ResourceString()),
+							Suggestion: osgraph.Suggestion(fmt.Sprintf("tag the image with hexadecimal ID %s into the image stream %s", imageID, imageStreamNode.ResourceString())),
+						})
+					}
+					continue
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.DockerImageRepositoryNode); ok {
```

Comment:
A From reference that is a docker pull spec(type DockerImage) isn't referencing an image stream at all, is my point, so what ImageStream would you be checking?  I don't think you'd ever get to this code path because `g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind)` should not return any results for an istNode of type DockerImageRepositoryNode.

At least that is my supposition, confirmation from someone who knows better (or testing it) would be good, but basically i'm saying i think this check is dead code.